aten/src/ATen/native/native_functions.yaml

```diff
@@ -7076,6 +7076,11 @@
   dispatch:
     CUDA: _scaled_grouped_mm_cuda
 
+- func: _grouped_mm(Tensor self, Tensor mat2, Tensor? offs=None, Tensor? bias=None, ScalarType? out_dtype=None) -> Tensor
```

Comment:
should we remove the out_dtype arg?

Reply:
I want to leave it for when we support fp32 out (many people want bf16-bf16 -> fp32 matmuls)

Reply:
That makes sense, I want to start to think about what the ideal lower level mm public primitives should be. We have allowed the `out_dtype` arg for scaled_mm and here here instead of the HOP dtype based out_dtype since these are private. I think that makes sense in the context of this private op and moving fast but when we ultimately make this and friends public I think we should have a unified way to do this 

cc @albanD

Reply:
how would HOP-based out_dtype look like?

Reply:
So we actually already have this: https://github.com/pytorch/pytorch/blob/3ac5a499ddac701f607a9f7206f9bec8871e1cbb/torch/_higher_order_ops/out_dtype.py#L31

AFAIK it does this by epilogue fusing the cast which ends up be a precision preserving since the compute dtype typically higher precision.

But I dont think this is well supported, garnered much traction..

Reply:
From what it looks to me, it will upcast bf16 output from the op into fp32, that's not what people want.

Reply:
Yeah I think it only has the desired effect in the context of compile, where any upcasts are fused into the triton impl thats generated. Agree that people will likely want a way to do this in eager

Reply:
And even in compile it'll break when using the config to preserve eager accuracy (which is used sometimes)? So yeah I'd rather have an eager implementation, but there's a question of how to handle backward for it - downcast the gradient (seems best, but loses accuracy) or perform backward compute in fp32 (unexpected slowdown)